### ISSUE TITLE
CentOS 7 breakage

### ISSUE THREAD

**Issue by Paladin** [schubergphilis/mod_security#32]
Just tried using this to install on a vagrant instance of CentOS7.0 and had (at least) two issues (I stopped after the second one appeared):

mod-security.load pointed to the wrong place for the module. Apparently at the time of writing the template "libdir" was null. So I manually set it (via the node[apache][libexecdir]) before including the recipe, and converged again.

The second time "http -M" failed under every id except root, because the none of the rules dirs apart from activated_rules allow pass-through for anyone but root (and since the activated_rules are links, it means none of the activated rules can be seen).

Not sure if the second one is properly your issue, but thought you should know about it, as well as do some controlling for libexecdir from apache2 being null

**Comment by mwijers** [comment]
This is already fixed in the latest version, 0.2.2. Did you use that one?

**Comment by Paladin** [comment]
Hmmm. Perhaps not. I'm pulling in the master branch from the github repo, rather than a version number, so it may not have updated since I first berk-ed it. I'll go take a second look.

Did a berks update of the github, didn't change the  .load file problem I mentioned. I'll switch this over to the supermarket version and try again.

**Comment by Paladin** [comment]
OK, tried the supermarket version, too, just in case there was a difference. Result:

The permissions on the rules directories was fixed in both, but the module never loaded because mods-enabled/mod-security.load still points by default to the wrong place:

LoadModule security2_module /mod_security2.so

The module is located with all the other apache modules, in /etc/httpd/modules/ which is getting lost somewhere along the way. It appears the template's libdir variable is empty. I can artificially force this by setting the apache2 value, but it may be a symptom of something getting lost between the two recipes, I haven't had the time to walk it down, yet (I'll worry about it after I get the full server build working properly, using the manual setting for now).

**Comment by MrSeccubus** [comment]
We had the exact same issue somewhere else. Old cookbook had an error it used the wrong variable. When you upload the new cookbook the apache cookbook first tries to start apache before the file is fixed. 
Startimg apache fails due to the bad file and breaks the chef run, thus file never gets fixed. 

I'll look at your suggestion for a safeguard.

**Comment by Paladin** [comment]
Could be a similar cause, but this is while running chefdk using chef-zero rather than a 'real' chef server.

**Comment by MrSeccubus** [comment]
Did you include the apache2 cookbook (before or after mod_security?)
The node attribute should come from there.

Maybe it is a good idea if we include it by default before we generate the file.

**Comment by Paladin** [comment]
Sorry about the delay in responding. Things got hectic. Yes, I was including apache2 before mod_security. I just built a fresh server, no workarounds, using the supermarket version(2.2). Tried to make everything start clean and fresh, and:

I ended up with two files to load the module in mods-available: security2.load and mod-security.load. The first one points to the correct location, the second does not. The conf file is mod-security.conf.

from the log output it seems to be the install_base_apache recipe from the mod_security cookbook that is writing the mod-security.load file (the wrong data) right after the unique_id module. I don't see a log entry for writing the security2.load file (the one with the correct data).
